### PR TITLE
prometheus prototype

### PR DESCRIPTION
**Pull request by kunmingg**
Provide stackdriver compatible prometheus ks prototype to support monitoring within kubeflow.

For https://github.com/kubeflow/kubeflow/issues/1160

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1204)

### CODE REVIEW

**Comment by kunmingg** [comment]
@lluunn 
As a prototype yes, but it haven't integrate with our services yet

File: kubeflow/core/prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+# * `prometheus.io/path`: If the metrics path is not `/metrics` override this.
+# * `prometheus.io/port`: If the metrics are exposed on a different port to the
+# service then set this appropriately.
+- job_name: 'kubernetes-service-endpoints'
+
+  kubernetes_sd_configs:
+  - role: endpoints
+
+  relabel_configs:
+#  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape]
```
Comment (lluunn) [inline kubeflow/core/prometheus.yml]: Is this intended to leave here or should be removed?

File: kubeflow/core/prometheus.yml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    action: replace
+    target_label: __metrics_path__
+  - source_labels: [__address__,__meta_kubernetes_service_annotation_prometheus_io_port]
+    action: replace
+    target_label: __address__
+    regex: (.+)(?::\d+);(\d+)
+    replacement: $1:$2
+
+remote_write:
+- url: "https://monitoring.googleapis.com:443/"
```
Comment (lluunn) [inline kubeflow/core/prometheus.yml]: what's this url?

File: kubeflow/core/prometheus.yml (hunk 1 above)
Comment (kunmingg) [reply kubeflow/core/prometheus.yml]: I think we might need to uncomment it and use as whitelist flag. 
And only scrape on signals we care about.

File: kubeflow/core/prometheus.yml (hunk 2 above)
Comment (kunmingg) [reply kubeflow/core/prometheus.yml]: We write metrics data to stackdriver through this url.

**Comment by kunmingg** [comment]
/hold
Need more change

**Comment by lluunn** [comment]
jsonnet format failed

**Pull request merged**